react/AddToCartButton.tsx

```diff
@@ -133,6 +142,16 @@ function AddToCartButton(props: Props) {
     event.stopPropagation()
     event.preventDefault()
 
+    if (onClickBehavior === "go-to-product-page") {
+      return navigate({
+        page: 'store.product',
+        params: {
+          slug: productLink?.linkText,
+          id: productLink?.productId,
```

Comment:
`productLink` is never `undefined`

```suggestion
          slug: productLink.linkText,
          id: productLink.productId,
```